src/rbac/default_role_manager.rs

```diff
@@ -0,0 +1,353 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+use std::collections::HashMap;
+use crate::rbac::RoleManager;
+
+type MatchingFunc = fn(&str, &str) -> bool;
+
+#[derive(Clone)]
+pub struct DefaultRoleManager {
+    pub all_roles: HashMap<String, Rc<RefCell<Role>>>,
+    pub max_hierarchy_level: usize,
+    pub has_pattern: bool,
+    pub matching_func: Option<MatchingFunc>,
+}
+
+impl DefaultRoleManager {
+    pub fn new(max_hierarchy_level: usize) -> Self {
+        return DefaultRoleManager {
+            all_roles: HashMap::new(),
+            max_hierarchy_level,
+            has_pattern: false,
+            matching_func: None,
+        };
+    }
+
+    fn create_role(&mut self, name: &str) -> Rc<RefCell<Role>> {
+        self.all_roles
+            .entry(name.to_owned())
+            .or_insert(Rc::new(RefCell::new(Role::new(name.to_owned()))))
+            .clone()
+    }
+
+    fn has_role(&self, name: &str) -> bool {
```

Comment:
This method implementation can be replaced with a call to `contains_key` on the underlying `HashMap`.

Reply:
Thanks, `contains_key` is better here, I will fix this later.